app/src/main/java/com/xffffff/wellfed/recipe/RecipeController.java

```diff
@@ -57,7 +57,10 @@ public void editRecipe(Recipe recipe) {
      */
     public void deleteRecipe(Recipe recipe) {
         recipeDB.delRecipe(recipe, (deleted, success) -> {
-            if (!success) {
+            if (success == null) {
```

Comment:
can we just return `deleted = null` and `success = false` as a response from the db, it would make more sense.

Reply:
Actually, I think it should be `(recipe, null)` so we can do `Cannot delete recipe " + recipe.getTitle() + " that is " +
                        "referenced by a meal."` It's hard to have it make sense where there are 3 situations,
                        

1. deletion failure
2. deletion failure due to referencing
3. success

Reply:
yeah , i think it's just a matter of preference with these 3 cases so i will mark it as resolved.